react/src/components/VFolderPermissionTag.tsx

```diff
@@ -0,0 +1,60 @@
+import DoubleTag, { DoubleTagObjectValue } from './DoubleTag';
+import { VFolderPermissionTag_VFolder$key } from './__generated__/VFolderPermissionTag_VFolder.graphql';
+import graphql from 'babel-plugin-relay/macro';
+import _ from 'lodash';
+import React from 'react';
+import { useFragment } from 'react-relay';
+
+const hasPermission = (permission: string | undefined, perm: string) => {
+  if (permission?.includes(perm)) {
+    return true;
+  }
+  if (permission?.includes('w') && perm === 'r') {
+    return true;
+  }
+  return false;
+};
+
+type VFolderPermissionTagProps =
+  | {
+      vFolderFrgmt?: never;
+      permission: string;
+    }
+  | {
+      vFolderFrgmt: VFolderPermissionTag_VFolder$key;
+      permission?: never;
+    };
+
+const VFolderPermissionTag: React.FC<VFolderPermissionTagProps> = ({
+  vFolderFrgmt = null,
+  permission,
+}) => {
+  const vFolder = useFragment(
```

Comment:
To use this component in other proejct, such as FastTrack, it still have to depend on `VirtualFolder`.